app/src/main/java/com/example/android/dessertpusher/MainActivity.kt

```diff
@@ -76,27 +68,51 @@ class MainActivity : AppCompatActivity(), LifecycleObserver {
         // Use Data Binding to get reference to the views
         binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
 
+        val thisApplication = requireNotNull(this).application
+        val dataSource = DataBase.getInstance(thisApplication)!!.databaseDao
+        val mainViewModelFactory = MainViewModelFactory(dataSource)
+
+        viewModel = ViewModelProvider(this, mainViewModelFactory).get(MainViewModel::class.java)
+
+        viewModel.getData()
+        binding.revenue = viewModel.revenue
+        binding.amountSold = viewModel.dessertsSold
+
+        val setTimer = object: CountDownTimer(1500, 100){
+            override fun onFinish() {
+                viewModel.getData()
+                binding.revenue = viewModel.revenue
+                binding.amountSold = viewModel.dessertsSold
+                showCurrentDessert()
+            }
+
+            override fun onTick(millisUntilFinished: Long) {
+                viewModel.getData()
+                binding.revenue = viewModel.revenue
+                binding.amountSold = viewModel.dessertsSold
+                showCurrentDessert()
+            }
+        }
+
+        setTimer.start()
```

Comment:
Could not find another way to refresh the data stored in the binding variables in the xml layout so I used this countdown timer. Hope you could show me a better way